### PR TITLE
util.TrimDomainName() fails when origin doesn't end in dot

### PR DESCRIPTION
**Pull request by tlimoncelli**
Fixes https://github.com/miekg/dns/issues/555

### CODE REVIEW

**Comment by miekg** [comment]
Thanks!

**Comment by tlimoncelli** [comment]
I made a couple more small changes.  PTAL

File: dnsutil/util.go [hunk 1]
```diff
@@ -36,7 +38,7 @@ func AddOrigin(s, origin string) string {
 	}
 
 	if origin == "." {
-		return s + origin // AddOrigin(s, ".") is an expensive way to add a ".".
+		return s + origin // AddOrigin(s, ".") is an easy way to add a "." without creating "..".
```
Comment (miekg) [inline dnsutil/util.go]: Just drop this comment. Doesn't add anything imo

File: dnsutil/util.go [hunk 2]
```diff
@@ -57,8 +59,12 @@ func TrimDomainName(s, origin string) string {
 		return strings.TrimSuffix(s, origin)
 	}
 
+	original_s := s
+	s = AddOrigin(s, ".")
```
Comment (miekg) [inline dnsutil/util.go]: you prolly can't upcall to dns.Fqdn() which is basically what this does?

File: dnsutil/util.go [hunk 3]
```diff
@@ -57,8 +59,12 @@ func TrimDomainName(s, origin string) string {
 		return strings.TrimSuffix(s, origin)
 	}
 
+	original_s := s
```
Comment (miekg) [inline dnsutil/util.go]: Don't use underscores in Go names.

**Comment by miekg** [comment]
Can you copy the .travis.yml changes from https://github.com/miekg/dns/pull/558/files ?

File: dnsutil/util.go (hunk 2 above)
Comment (tlimoncelli) [reply dnsutil/util.go]: Done

File: dnsutil/util.go (hunk 3 above)
Comment (tlimoncelli) [reply dnsutil/util.go]: Done

**Pull request merged**